Headers/Foundation/NSOrderedSet.h

```diff
@@ -0,0 +1,212 @@
+- (instancetype) initWithOrderedSet:(GS_GENERIC_CLASS(NSOrderedSet, ElementT)*)aSet;
+- (instancetype) initWithOrderedSet:(GS_GENERIC_CLASS(NSOrderedSet, ElementT)*)aSet
+                          copyItems:(BOOL)flag;
+- (instancetype) initWithOrderedSet:(GS_GENERIC_CLASS(NSOrderedSet, ElementT)*)aSet
+                              range: (NSRange)range
+                          copyItems:(BOOL)flag;
+- (instancetype) initWithSet:(GS_GENERIC_CLASS(NSSet, ElementT)*)aSet;
+- (instancetype) initWithSet:(GS_GENERIC_CLASS(NSSet, ElementT)*)aSet copyItems:(BOOL)flag;
+- (instancetype) init;
+- (NSUInteger) count;
+- (BOOL)containsObject:(GS_GENERIC_TYPE(ElementT))anObject;
+- (void) enumerateObjectsAtIndexes:(NSIndexSet *)indexSet
+                           options:(NSEnumerationOptions)opts
+                        usingBlock:(GSEnumeratorBlock)aBlock;
+- (void) enumerateObjectsUsingBlock: (GSEnumeratorBlock)aBlock;
+- (void) enumerateObjectsWithOptions:(NSEnumerationOptions)opts
+                          usingBlock:(GSEnumeratorBlock)aBlock;
+- (GS_GENERIC_TYPE(ElementT)) firstObject;
+- (GS_GENERIC_TYPE(ElementT)) lastObject;
+- (GS_GENERIC_TYPE(ElementT)) objectAtIndex: (NSUInteger)index;
+  - (GS_GENERIC_TYPE(ElementT)) objectAtIndexedSubscript:(NSUInteger)index;
+- (GS_GENERIC_CLASS(NSArray, ElementT)*) objectsAtIndexes:(NSIndexSet *)indexes;
+- (NSUInteger) indexOfObject:(GS_GENERIC_TYPE(ElementT))objects;  
+- (NSUInteger) indexOfObject: (id)key
+               inSortedRange: (NSRange)range
+                     options: (NSBinarySearchingOptions)options
+             usingComparator: (NSComparator)comparator;
+
+- (NSUInteger) indexOfObjectAtIndexes:(NSIndexSet *)indexSet
+                              options:(NSEnumerationOptions)opts
+                          passingTest:(GSPredicateBlock)predicate;
+- (NSUInteger) indexOfObjectPassingTest:(GSPredicateBlock)predicate;
+- (NSUInteger) indexOfObjectWithOptions:(NSEnumerationOptions)opts
+                            passingTest:(GSPredicateBlock)predicate;
+- (NSIndexSet *) indexesOfObjectsAtIndexes:(NSIndexSet *)indexSet
+                              options:(NSEnumerationOptions)opts
+                          passingTest:(GSPredicateBlock)predicate;
+
+- (NSIndexSet *)indexesOfObjectsPassingTest:(GSPredicateBlock)predicate;
+- (NSIndexSet *) indexesOfObjectsWithOptions:(NSEnumerationOptions)opts
+                            passingTest:(GSPredicateBlock)predicate;
+- (GS_GENERIC_CLASS(NSEnumerator, ElementT)*) objectEnumerator;
+- (GS_GENERIC_CLASS(NSEnumerator, ElementT)*) reverseObjectEnumerator;
+- (NSOrderedSet *)reversedOrderedSet;
+- (void) getObjects: (__unsafe_unretained GS_GENERIC_TYPE(ElementT)[])aBuffer
+              range: (NSRange)aRange;
+
+// Key value coding support
+- (void) setValue: (GS_GENERIC_TYPE(ElementT))value forKey: (NSString*)key;
+- (GS_GENERIC_TYPE(ElementT)) valueForKey: (NSString*)key; 
+
+// Key-Value Observing Support
+/*
```

Comment:
We do not need to declare these methods here but should we implement them? Apple documents that they raise an exception. Is this really true? It should be easy to implement them similar to NSArray. Why doesn't Apple want them to be there?

Reply:
For now I'm going to raise an exception as recommended in the docs, but I believe we should revisit this going forward.  It doesn't really make any sense why they don't want this to be observable.

Reply:
Added